app/src/main/java/com/riadh/movies/ui/fragments/SearchFragment.java

```diff
@@ -0,0 +1,203 @@
+import org.androidannotations.annotations.App;
+import org.androidannotations.annotations.Bean;
+import org.androidannotations.annotations.EFragment;
+import org.androidannotations.annotations.ViewById;
+
+import retrofit2.Response;
+
+@EFragment(R.layout.fragment_search)
+public class SearchFragment extends Fragment implements RecyclerViewItemSimpleClick<Result> {
+
+    @ViewById(R.id.progress_view)
+    CircularProgressView progressView;
+
+    @ViewById(R.id.empty_movies)
+    View emptyView;
+
+    @ViewById(R.id.input_search)
+    EditText searchTxt;
+
+    @App
+    MyApplication app;
+
+    @ViewById(R.id.movies_list)
+    RecyclerViewEmptySupport dreamsList;
+
+    @ViewById(R.id.swipeRefreshLayout)
+    SwipeRefreshLayout swipeRefreshLayout;
+    @Bean
+    MoviesAdapter feedAdapter;
+    int pastVisibleItems, visibleItemCount, totalItemCount;
+    LinearLayoutManager lm;
+    String searchValue = "";
+    private MoviesResults moviesResults;
+    private boolean isCallingApi = false;
+
+    @AfterViews
+    void onScreenLoaded() {
+        ((BaseActivity) getActivity()).configActionBarWithTitle(R.string.search);
+        populateMovies();
+
+        searchTxt.setOnEditorActionListener(new TextView.OnEditorActionListener() {
+            @Override
+            public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
+                if (actionId == EditorInfo.IME_ACTION_SEARCH || actionId == EditorInfo.IME_ACTION_DONE || actionId == EditorInfo.IME_ACTION_NEXT) {
+                    searchValue = searchTxt.getText().toString();
+                    searchMovie(searchValue, 1);
+                    MyUtils.hideSoftKeyboard(v);
+                    return true;
+                }
+                return false;
+            }
+        });
+
+        searchTxt.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+
+            }
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+
+            }
+
+            @Override
+            public void afterTextChanged(Editable editable) {
+                if (editable.toString().equals("")) {
+                    searchValue = "";
+                }
+            }
+        });
+
+
+    }
+
+    private void populateMovies() {
+        setupPropertiesAdapter();
+    }
+
+    private void setupPropertiesAdapter() {
+
+        lm =
+                new LinearLayoutManager(getContext(), LinearLayoutManager.VERTICAL, false);
+        dreamsList.setLayoutManager(lm);
+        feedAdapter.setOnItemClickListener(this);
+
+        dreamsList.setAdapter(feedAdapter);
+        dreamsList.setEmptyView(emptyView);
+
+
+        swipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+            @Override
+            public void onRefresh() {
+                if (!searchValue.equals("")) {
```

Comment:
is this null safe?

Reply:
for this code would be useless... because it can't be null
searchValue = searchTxt.getText().toString(); this is the only way it's getting a value .
but if someone changes my code... indeed you are right and i should use 
TextUtils.isEmpty(searchValue)